### PR TITLE
🐛 Let `Curator.from_tiledbsoma()` validate and annotate all `.obs` columns

### PR DESCRIPTION
**Pull request by falexwolf**
Fixes:

- https://github.com/laminlabs/lamindb/issues/2285

### CODE REVIEW

File: lamindb/_curate.py [hunk 1]
```diff
@@ -1168,8 +1171,12 @@ def _check_save_keys(self):
         with _open_tiledbsoma(self._experiment_uri, mode="r") as experiment:
             experiment_obs = experiment.obs
             self._n_obs = len(experiment_obs)
-            valid_obs_keys = [k for k in experiment_obs.keys() if k != "soma_joinid"]
+            valid_obs_keys = [
+                k for k in experiment_obs.schema.names if k != "soma_joinid"
```
Comment (Koncopd) [inline lamindb/_curate.py:1175]: Please read from the saved schema here to avoid redundant requests, maybe they cache but let’s be careful anyways.

**Comment by Koncopd** [comment]
Thank you for fixing, this is great, just one small thing here to correct.

File: lamindb/_curate.py (hunk 1 above)
Comment (Koncopd) [reply lamindb/_curate.py:1175]: You save the schema on the next line anyways.

Comment (falexwolf) [reply lamindb/_curate.py]: Done!

**Pull request merged**